ui/app/config/views/logStoreForm.tsx

```diff
@@ -0,0 +1,149 @@
+"use client";
+
+import { Alert, AlertDescription } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Switch } from "@/components/ui/switch";
+import { useGetLogStoreConfigQuery, useUpdateLogStoreConfigMutation } from "@/lib/store";
+import { LogStoreConfig, SQLiteConfig } from "@/lib/types/config";
+import { getErrorMessage } from "@/lib/store";
+import { AlertTriangle, Database, Save } from "lucide-react";
+import { useCallback, useEffect, useState } from "react";
+import { toast } from "sonner";
+
+const defaultSQLiteConfig: SQLiteConfig = {
+	path: "./bifrost.db",
+};
+
+export default function LogStoreForm() {
+	const { data: logStoreConfig, isLoading } = useGetLogStoreConfigQuery();
+	const [updateLogStoreConfig] = useUpdateLogStoreConfigMutation();
+
+	const [localConfig, setLocalConfig] = useState<LogStoreConfig>({
+		enabled: false,
+		type: "sqlite",
+		config: defaultSQLiteConfig,
+	});
+
+	const [needsRestart, setNeedsRestart] = useState(false);
+	const [hasChanges, setHasChanges] = useState(false);
+
+	// Update local config when data is loaded
+	useEffect(() => {
+		if (logStoreConfig) {
+			setLocalConfig(logStoreConfig);
+			setHasChanges(false);
+		}
+	}, [logStoreConfig]);
+
+	// Track changes
+	useEffect(() => {
+		if (logStoreConfig) {
+			const hasConfigChanges = JSON.stringify(localConfig) !== JSON.stringify(logStoreConfig);
+			setHasChanges(hasConfigChanges);
+			setNeedsRestart(hasConfigChanges);
```

Comment:
won't this logic not show the alert when we make some change and the just reload the page without restart bifrost? as we are directly making update in the DB via the handlers